### ISSUE TITLE
Accessibility: error message and help text not read out by screenreader

### ISSUE THREAD

**Issue by geoffrich** [AlaskaAirlines/ods-inputtext#23]
## Describe the bug
When I focus ods-inputtext using a screenreader, the associated error message is not read out.

## To Reproduce
1. Render the following markup:
```html
<ods-inputtext label="Passenger's last name" type="text" required="" error="test error">
</ods-inputtext>
```
2. Load the page and focus the input
3. Notice that the error message is not read out. NVDA/Firefox reads out "Passenger's last name edit required invalid entry has auto-complete"

NVDA also fails to read out the help text, if set. I only tested with NVDA but my guess is this happens with other screen readers as well.

## Expected behavior
NVDA reads out the error message and/or help text when the form field is focused.

## Environment
OS: Windows 10
Browser: Firefox
Screen reader: NVDA

## Additional context
I don't think the aria-live regions we currently use in the component are working as expected. I think it would be better to link the error message and help text using aria-describedby.

**Comment by blackfalcon** [comment]
Additional ARIA tools were added to the component to bind the help text with the description of the interaction. Also updated the alert to be assertive when there is an error with the text field as to alert the user as soon as they exit a required field with an immediate alert. 

This was tested with Chrome and Edge using Apple Voice Over.
